api/src/auth/__tests__/auth.service.spec.ts

```diff
@@ -0,0 +1,301 @@
+import { beforeEach, describe, expect, it, jest } from 'bun:test';
+
+import type { CreateAccountDto } from '../dto/create-account.dto';
+import type { VerifyAccountDto } from '../dto/verify-account.dto';
+import type { CurrentUser, Locale } from '@databank/types';
+import { CryptoService } from '@douglasneuroinformatics/nestjs/modules';
+import { type MockedInstance, createMock } from '@douglasneuroinformatics/nestjs/testing';
+import { ForbiddenException, NotFoundException, UnauthorizedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { JwtService } from '@nestjs/jwt';
+import { Test } from '@nestjs/testing';
+
+import { I18nService } from '@/i18n/i18n.service';
+import { MailService } from '@/mail/mail.service';
+import { createUserDtoStubFactory } from '@/users/__tests__/stubs/create-user.dto.stub';
+import type { CreateUserDto } from '@/users/dto/create-user.dto';
+import type { User } from '@/users/schemas/user.schema';
+import { UsersService } from '@/users/users.service';
+
+import { AuthService } from '../auth.service';
+import { createAccountDtoStubFactory } from './stubs/create-account.dto.stub';
+import { currentUserStubFactory } from './stubs/current-user.stub';
+import { verifyAccountStubFactory } from './stubs/verify-account.dto.stub';
+
+const validDate = Date.now() + 360000;
+const expiredDate = Date.now() - 10000;
+
+describe('AuthService', () => {
+  let authService: AuthService;
+  let usersService: MockedInstance<UsersService>;
+  let cryptoService: MockedInstance<CryptoService>;
+  let jwtService: MockedInstance<JwtService>;
+
+  // called before every test to intialize the AuthService
+  beforeEach(async () => {
+    const moduleRef = await Test.createTestingModule({
+      providers: [
+        AuthService,
+        {
+          provide: ConfigService,
+          useValue: {
+            getOrThrow: (propertyPath: string) => propertyPath
+          }
+        },
+        {
+          provide: CryptoService,
+          useValue: createMock(CryptoService)
+        },
+        {
+          provide: I18nService,
+          useValue: createMock(I18nService)
+        },
+        { provide: JwtService, useValue: createMock(JwtService) },
+        {
+          provide: MailService,
+          useValue: createMock(MailService)
+        },
+        {
+          provide: UsersService,
+          useValue: createMock(UsersService)
+        }
+      ]
+    }).compile();
+    // get all the services we need from the moduleRef
+    authService = moduleRef.get(AuthService);
+    usersService = moduleRef.get(UsersService);
+    cryptoService = moduleRef.get(CryptoService);
+    jwtService = moduleRef.get(JwtService);
+  });
+
+  it('should be defined', () => {
+    expect(authService).toBeDefined();
+  });
+
+  let createAccountDto: CreateAccountDto;
+  let createUserDto: CreateUserDto;
+  let verifyAccountDto: VerifyAccountDto;
+  let currentUser: CurrentUser;
+
+  beforeEach(() => {
```

Comment:
there should not be two `beforeEach` blocks at the same level of nesting. Move this into the previous block